usage/usage.go

```diff
@@ -0,0 +1,136 @@
+// Package usage implements usage tracking for k6 in order to figure what is being used within a given execution
+package usage
+
+import (
+	"strings"
+	"sync"
+)
+
+// Usage is a way to collect usage data for within k6
+type Usage struct {
+	l *sync.Mutex
+	m map[string]any
+}
+
+// New returns a new empty Usage ready to be used
+func New() *Usage {
+	return &Usage{
+		l: new(sync.Mutex),
+		m: make(map[string]any),
+	}
+}
+
```

Comment:
I started with the idea of either reusing [expvar](https://pkg.go.dev/expvar) or the new [golang telemetry](https://go.dev/doc/telemetry) , but both of those are global and I didn't really want to add global state to k6.

btw it seems at least loki does [use expvar](https://github.com/grafana/loki/blob/7c1a8493b3837396d085547a42d8260271f1d68a/pkg/analytics/stats.go#L210) , but I am not certain how much this can apply to us as well